### ISSUE TITLE
[jvm-packages] false should be the default value of maximize_evaluation_metrics

### ISSUE THREAD

**Issue by david-liu** [dmlc/xgboost#4422]
https://github.com/dmlc/xgboost/blob/253fdd8a42d5ec6b819788199584d27bf9ea6253/jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala#L161

I use spark `XGBoostRegressor` to train model and set the NumEarlyStoppingRounds and the metric is ` rmse`, but my job throw exception:
```
19/04/30 01:42:13 WARN TaskSetManager: Lost task 44.0 in stage 69.0 (TID 5371, hdc42-mcc10-01-1110-4204-067-tess0029.stratus.rno.ebay.com, executor 52): java.lang.IllegalArgumentException: maximize_evaluation_metrics has to be specified
```
which means the `XGBoostRegressor`  did not set the default value for `maximize_evaluation_metrics`. I think  the default value should be `false`, considering most matrics are decresing the loss, event the use did not set it (Except `AUC`)

**Comment by hcho3** [comment]
@david-liu This behavior is documented in https://xgboost.readthedocs.io/en/latest/jvm/xgboost4j_spark_tutorial.html#early-stopping. I do agree that automatically setting `maximize_evaluation_metrics` would be a nice feature to have. Feel free to submit a pull request.
